### PR TITLE
fix: clear dashboard storage if uuids dont match

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: [#11018](https://github.com/lightdash/lightdash/issues/11018)

### Description:

Replication steps:
- go to dashboard A
- click edit chart
- change browser URL to dashboard B in edit mode

Before:

https://github.com/user-attachments/assets/e1c1a859-5388-4146-9bfc-0a28ab44deb9

After:

https://github.com/user-attachments/assets/2f3834eb-5529-4935-a133-24146a0c3dfb

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
Can you check if you can save a chart in a dashboard?

File: packages/frontend/src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -725,9 +725,20 @@ const Dashboard: FC = () => {
...
     const { user } = useApp();
     const dashboardCommentsCheck = useDashboardCommentsCheck(user?.data);
+    const { clearDashboardStorage, getDashboardUuid } = useDashboardStorage();
+
+    useEffect(() => {
+        if (dashboardUuid !== getDashboardUuid()) {
+            // Clear storage if the dashboardUuid in storage doesn't match the current dashboardUuid
+            clearDashboardStorage();
+        }
+    }, [dashboardUuid, clearDashboardStorage, getDashboardUuid]);
```
Comment (joaoviana) [inline packages/frontend/src/pages/Dashboard.tsx:741]: this prevents us from saving charts within dashboards. Perhaps a different check needs to be done?

Comment (owlas) [reply packages/frontend/src/pages/Dashboard.tsx:741]: what about keying the storage on the dashboardUuid? So you have a cache per-id?

Comment (ZeRego) [reply packages/frontend/src/pages/Dashboard.tsx:741]: We would still need to store the dashboardUuid separately so when we go from dashboard to create chart it knows from what cache to fetch. I propose we close this ticket as it is an edge case, we stopped deleting the charts in the DB and we are planning to change this UI flow.

**Pull request closed without merging**